sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/ManagedServiceIdentity.Serialization.cs

```diff
@@ -262,7 +262,12 @@ internal static ManagedServiceIdentity DeserializeManagedServiceIdentity(JsonEle
                     Dictionary<ResourceIdentifier, UserAssignedIdentity> dictionary = new Dictionary<ResourceIdentifier, UserAssignedIdentity>();
                     foreach (var property0 in property.Value.EnumerateObject())
                     {
-                        dictionary.Add(new ResourceIdentifier(property0.Name), ModelReaderWriter.Read<UserAssignedIdentity>(new BinaryData(Encoding.UTF8.GetBytes(property0.Value.GetRawText())), options, AzureResourceManagerContext.Default));
+#if NET9_0_OR_GREATER
```

Comment:
While it's a fine change, be aware that this path will not trigger until .NET 10 is released and we move library target frameworks to the new LTS.